framework/src/main/java/org/tron/core/metrics/blockchain/BlockChainMetricManager.java

```diff
@@ -0,0 +1,217 @@
+        WitnessInfo witness = new WitnessInfo(address,
+                block.getInstance().getBlockHeader().getRawData().getVersion());
+        witnessInfos.add(witness);
+      }
+    }
+    return witnessInfos;
+  }
+
+  private RateInfo getBlockProcessTime() {
+    RateInfo blockProcessTime = new RateInfo();
+    blockProcessTime.setCount(metricsService.getMeter(MetricsKey.BLOCKCHAIN_BLOCK_COUNT)
+            .getCount());
+    blockProcessTime.setMeanRate(getAvgBlockProcessTimeByGap(0));
+    blockProcessTime.setOneMinuteRate(getAvgBlockProcessTimeByGap(1));
+    blockProcessTime.setFiveMinuteRate(getAvgBlockProcessTimeByGap(5));
+    blockProcessTime.setFifteenMinuteRate(getAvgBlockProcessTimeByGap(15));
+    return blockProcessTime;
+  }
+
+  private RateInfo getTransactionRate() {
+    Meter transactionRate = metricsService.getMeter(MetricsKey.BLOCKCHAIN_TPS);
+    RateInfo tpsInfo = new RateInfo();
+    tpsInfo.setCount(transactionRate.getCount());
+    tpsInfo.setMeanRate(transactionRate.getMeanRate());
+    tpsInfo.setOneMinuteRate(transactionRate.getOneMinuteRate());
+    tpsInfo.setFiveMinuteRate(transactionRate.getFiveMinuteRate());
+    tpsInfo.setFifteenMinuteRate(transactionRate.getFifteenMinuteRate());
+    return tpsInfo;
+  }
+
+
+  // gap: 1 minute, 5 minute, 15 minute, 0: avg for total block and time
+  private double getAvgBlockProcessTimeByGap(int gap) {
+    Meter meterBlockProcessTime =
+        metricsService.getMeter(MetricsKey.BLOCKCHAIN_BLOCKPROCESS_TIME);
+    Meter meterBlockTxCount = metricsService.getMeter(MetricsKey.BLOCKCHAIN_BLOCK_COUNT);
+    if (meterBlockTxCount.getCount() == 0) {
+      return 0;
+    }
+    switch (gap) {
+      case 0:
+        return (meterBlockProcessTime.getCount() / (double) meterBlockTxCount.getCount());
+      case 1:
+        int gapMinuteTimeBlock =
+            Math.round(Math.round(meterBlockProcessTime.getOneMinuteRate() * 60));
+        int gapMinuteCount = Math.round(Math.round(meterBlockTxCount.getOneMinuteRate() * 60));
+        return gapMinuteTimeBlock / (double) gapMinuteCount;
+      case 5:
+        int gapFiveTimeBlock =
+            Math.round(Math.round(meterBlockProcessTime.getFiveMinuteRate() * gap * 60));
+        int gapFiveTimeCount =
+            Math.round(Math.round(meterBlockTxCount.getFiveMinuteRate() * gap * 60));
+        return gapFiveTimeBlock / (double) gapFiveTimeCount;
+      case 15:
+        int gapFifteenTimeBlock =
+            Math.round(Math.round(meterBlockProcessTime.getFifteenMinuteRate() * gap * 60));
+        int gapFifteenTimeCount =
+            Math.round(Math.round(meterBlockTxCount.getFifteenMinuteRate() * gap * 60));
+        return gapFifteenTimeBlock / (double) gapFifteenTimeCount;
+
+      default:
+        return -1;
+    }
+  }
+
+  private int getSuccessForkCount() {
+    return (int) metricsService.getMeter(MetricsKey.BLOCKCHAIN__FORK_COUNT).getCount();
+  }
+
+  private int getFailForkCount() {
```

Comment:
Use public methods, no need to encapsulate

Reply:
ok